slick-testkit/src/main/scala/com/typesafe/slick/testkit/tests/ActionTest.scala

```diff
@@ -218,4 +218,109 @@ class ActionTest extends AsyncTest[RelationalTestDB] {
+        case Failure(e: slick.SlickException) if tdb.profile == slick.memory.MemoryProfile => ()
+        case Failure(e) => throw e
+        case Success(_) => throw new Exception("Should have failed to create new table. Table exists")
+      }
+      _ <- ts2.schema.createIfNotExists
+      _ <- ts2.schema.create.asTry.map{
+        case Failure(e:java.sql.SQLException) if e.getMessage.toLowerCase.contains("""already exists""") => ()
+        case Failure(e: slick.SlickException) if tdb.profile == slick.memory.MemoryProfile => ()
+        case Failure(e) => throw e
+        case Success(_) => throw new Exception("Should have failed to create new table. Table exists")
+      }
+      initial3 <- ts.result
+      _ = assert(initial3.toSet == Set())
+      initial4 <- ts2.result
+      _ = assert(initial4.toSet == Set())
+      _ <- ts.schema.dropIfExists
+      _ <- ts.schema.drop.asTry.map{
+        case Failure(e:java.sql.SQLException) => ()
+        case Failure(e: slick.SlickException) if tdb.profile == slick.memory.MemoryProfile => ()
+        case Failure(e) => throw e
+        case Success(_) => throw new Exception("Should have failed to drop table. Table already dropped")
+      }
+      _ <- ts2.schema.dropIfExists
+      _ <- ts2.schema.drop.asTry.map{
+        case Failure(e:java.sql.SQLException) => ()
+        case Failure(e: slick.SlickException) if tdb.profile == slick.memory.MemoryProfile => ()
+        case Failure(e) => throw e
+        case Success(_) => throw new Exception("Should have failed to drop table. Table already dropped")
+      }
+      //test batch create/drop
+      _ <- batch.create
+      _ <- batch.createIfNotExists
+      res <- (ts ++= Seq(2, 3, 1, 5, 4)) >> ts.result
+      _ = assert(res.toSet == Set(2, 3, 1, 5, 4))
+      res2 <- (ts2 ++= Seq("2", "3", "1", "5", "4")) >> ts2.result
+      _ = assert(res2.toSet == Set("2", "3", "1", "5", "4"))
+      _ <- batch.drop
+      _ <- batch.dropIfExists
+      _ <- batch.createIfNotExists
+      _ <- batch.create.asTry.map{
+        case Failure(e:java.sql.SQLException) => ()
+        case Failure(e: slick.SlickException) if tdb.profile == slick.memory.MemoryProfile => ()
+        case Failure(e) => throw e
+        case Success(_) => throw new Exception("Should have failed to create new table. Table exists")
+      }
+      _ <- batch.dropIfExists
+      _ <- batch.drop.asTry.map{
+        case Failure(e:java.sql.SQLException) => ()
+        case Failure(e: slick.SlickException) if tdb.profile == slick.memory.MemoryProfile => ()
+        case Failure(e) => throw e
+        case Success(_) => throw new Exception("Should have failed to drop table. Table already dropped")
+      }
+      //test create/drop with constraints
+      _ <- as.schema.createIfNotExists
+      _ <- as ++= Seq((1, 1, "a11"), (1, 2, "a12"), (2, 1, "a21"), (2, 2, "a22") )
+      _ <- (as += (1, 1, "a11-conflict")).failed
+      _ <- as.schema.drop
+    } yield ()
+  }
```

Comment:
@trevorsibanda maybe you can split this test into multiple smaller ones, so it will be easier to narrow down where stuff goes wrong? Unfortunately both future and DBIO kill the usefulness of stacktraces